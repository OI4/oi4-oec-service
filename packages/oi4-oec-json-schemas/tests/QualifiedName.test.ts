import { matchersWithOptions } from 'jest-json-schema'
import schema from '../src/schemas/QualifiedName.schema.json'
import uint16 from '../src/schemas/dataTypes/uint16.schema.json'

expect.extend(
  matchersWithOptions({
    // Loading in a schema which is comprised only of definitions,
    // which means specific test schemas need to be created.
    // This is good for testing specific conditions for definition schemas.
    schemas: [uint16],
    verbose: true,
  })
)

describe('QualifiedName schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it('test valid values', () => {
    expect({ name: '', namespaceIndex: 0 }).toMatchSchema(schema)
    expect({ name: '1', namespaceIndex: 0 }).toMatchSchema(schema)
    expect({ name: 'a', namespaceIndex: 0 }).toMatchSchema(schema)
    expect({ name: 'b', namespaceIndex: 65534 }).toMatchSchema(schema)
    expect({ name: 'c', namespaceIndex: 65535 }).toMatchSchema(schema)
  })

  it('test invalid values', () => {
    const invalidValues = [
      { name: 1, namespaceIndex: 1 },
      { name: 'b', namespaceIndex: 65536 },
      { name: 'c', namespaceIndex: -1 },
      { name: 'c', namespaceIndex: '1' },
    ]

    invalidValues.forEach((value, index) => {
      expect(value).not.toMatchSchema(schema)
      expect(() => {
        expect(value).toMatchSchema(schema)
      }).toThrowErrorMatchingSnapshot(index.toString())
    })
  })
})
