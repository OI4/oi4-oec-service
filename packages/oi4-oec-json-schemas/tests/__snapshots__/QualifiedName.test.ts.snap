// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QualifiedName schema test invalid values: 0 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'Name'
    Path:     QualifiedName.schema.json#/required
  must have required property 'NamespaceIndex'
    Path:     QualifiedName.schema.json#/required
  must NOT have additional properties, but found 'name'
    Expected: <boolean> false
    Path:     QualifiedName.schema.json#/additionalProperties
  must NOT have additional properties, but found 'namespaceIndex'
    Expected: <boolean> false
    Path:     QualifiedName.schema.json#/additionalProperties
"
`;

exports[`QualifiedName schema test invalid values: 1 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'Name'
    Path:     QualifiedName.schema.json#/required
  must have required property 'NamespaceIndex'
    Path:     QualifiedName.schema.json#/required
  must NOT have additional properties, but found 'name'
    Expected: <boolean> false
    Path:     QualifiedName.schema.json#/additionalProperties
  must NOT have additional properties, but found 'namespaceIndex'
    Expected: <boolean> false
    Path:     QualifiedName.schema.json#/additionalProperties
"
`;

exports[`QualifiedName schema test invalid values: 2 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'Name'
    Path:     QualifiedName.schema.json#/required
  must have required property 'NamespaceIndex'
    Path:     QualifiedName.schema.json#/required
  must NOT have additional properties, but found 'name'
    Expected: <boolean> false
    Path:     QualifiedName.schema.json#/additionalProperties
  must NOT have additional properties, but found 'namespaceIndex'
    Expected: <boolean> false
    Path:     QualifiedName.schema.json#/additionalProperties
"
`;

exports[`QualifiedName schema test invalid values: 3 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'Name'
    Path:     QualifiedName.schema.json#/required
  must have required property 'NamespaceIndex'
    Path:     QualifiedName.schema.json#/required
  must NOT have additional properties, but found 'name'
    Expected: <boolean> false
    Path:     QualifiedName.schema.json#/additionalProperties
  must NOT have additional properties, but found 'namespaceIndex'
    Expected: <boolean> false
    Path:     QualifiedName.schema.json#/additionalProperties
"
`;
